packages/compiler/core/types.ts

```diff
@@ -958,20 +941,17 @@ export interface LineAndCharacter {
   character: number;
 }
 
-export interface JsSourceFile {
-  readonly kind: "JsSourceFile";
+export interface JsSourceFileNode extends DeclarationNode, BaseNode {
+  readonly kind: SyntaxKind.JsSourceFile;
 
   /* A source file with empty contents to represent the file on disk. */
   readonly file: SourceFile;
 
   /* The exports object as comes from `import()` */
   readonly esmExports: any;
 
-  /* Exported "global scope" bindings */
-  exports?: SymbolTable<DecoratorSymbol | FunctionSymbol>;
-
   /* Any namespaces declared by decorators. */
-  readonly namespaces: readonly NamespaceStatementNode[];
+  readonly namespaceSymbols: Sym[];
```

Comment:
Should the `CadlScriptNode` also have a similar change?

Reply:
Yeah not a bad idea, looks like usages of this pretty much immediately grab the symbol (except for the case where we create the type).

Reply:
On reflection I've left this as is, it was sort of a half-dozen-of-one-six-of-the-other situation. CadlScriptNode's namespaces are actually real namespaces though, unlike the JS files ones which don't exist.